src/Strategy/AlpineLinux/Parser.hs

```diff
@@ -0,0 +1,128 @@
+-- | Parsing functions for APK files.
+-- APK is the package manager for Alpine Linux (https://alpinelinux.org/).
+--
+-- The database itself is just a flat file with entries in text.  These are
+-- specified in a key-value format with obscure names.  See
+-- https://wiki.alpinelinux.org/wiki/Apk_spec#Syntax.
+module Strategy.AlpineLinux.Parser (installedPackagesDatabaseParser, PackageError (..)) where
+
+import Data.List.NonEmpty (NonEmpty (..))
+import Data.List.NonEmpty qualified as NE
+import Data.Map qualified as Map
+import Data.String.Conversion (toText)
+import Data.Text (Text)
+import Data.Void (Void)
+import Strategy.AlpineLinux.Types (AlpinePackage (..))
+import Text.Megaparsec (
+  MonadParsec (eof, takeWhileP),
+  Parsec,
+  many,
+  optional,
+  some,
+  try,
+ )
+import Text.Megaparsec.Char (char, eol, letterChar)
+
+-- | Errors that can occur even in a valid file.  Packages are a lose collection
+-- of fields, but it unclear which are required.  These error represent fields
+-- we expect, but which may be missing in unusual circumstances.
```

Comment:
[nit]: typo and wording
```suggestion
-- | Errors from file with valid syntax but invalid data.  Packages are a loose collection
-- of fields, but it is unclear which are required.  These errors represent fields
-- we expect, but which may be missing in unusual circumstances.
```

Reply:
I spend so much time trying to remember to write "lose" instead of "loose" that I made the opposite mistake here. :joy: